### ISSUE TITLE
`WordPress.WP.CapitalPDangit` overly strict and can correct system variables.

### ISSUE THREAD

**Issue by peterwilsoncc** [WordPress-Coding-Standards/WordPress-Coding-Standards#1698]
## Bug Description

The ``WordPress.WP.CapitalPDangit` sniff is overly strict and can correct definitions such as database usernames and other system settings.

### Before running phpcbf
```php
<?php
define( 'DB_NAME', 'wordpress' );
define( 'DB_USER', 'wordpress' );
define( 'WP_TESTS_DOMAIN', 'wordpress.example.org' );
```

### After running phpcbf

```php
<?php
define( 'DB_NAME', 'WordPress' );
define( 'DB_USER', 'WordPress' );
define( 'WP_TESTS_DOMAIN', 'WordPress.example.org' );
```

## Error Code

Code run in command line:

phpcs report
```
./vendor/bin/phpcs --standard=phpcs.xml.dist  --report=full sniff-test.php
W 1 / 1 (100%)

FILE: sniff-test.php
----------------------------------------------------------------------------------------------------------------------------------
FOUND 0 ERRORS AND 3 WARNINGS AFFECTING 3 LINES
----------------------------------------------------------------------------------------------------------------------------------
 2 | WARNING | [x] Please spell "WordPress" correctly. Found 1 misspelling(s): wordpress
   |         |     (WordPress.WP.CapitalPDangit.Misspelled)
 3 | WARNING | [x] Please spell "WordPress" correctly. Found 1 misspelling(s): wordpress
   |         |     (WordPress.WP.CapitalPDangit.Misspelled)
 4 | WARNING | [x] Please spell "WordPress" correctly. Found 1 misspelling(s): wordpress
   |         |     (WordPress.WP.CapitalPDangit.Misspelled)
----------------------------------------------------------------------------------------------------------------------------------
PHPCBF CAN FIX THE 3 MARKED SNIFF VIOLATIONS AUTOMATICALLY
----------------------------------------------------------------------------------------------------------------------------------
```

phpcbf command
```
./vendor/bin/phpcbf --standard=phpcs.xml.dist  -s sniff-test.php
F 1 / 1 (100%)

PHPCBF RESULT SUMMARY
----------------------------------------------------------------------
FILE                                                  FIXED  REMAINING
----------------------------------------------------------------------
sniff-test.php                                        3      0
----------------------------------------------------------------------
A TOTAL OF 3 ERRORS WERE FIXED IN 1 FILE
----------------------------------------------------------------------
```

## Environment

| Question               | Answer
| ------------------------| -------
| PHP version             | 7.1.23 
| PHP_CodeSniffer version | 3.4.0
| WPCS version            | 2.0.0
| WPCS install type       | Composer project (WordPress Core) local

## Tested Against `develop` branch?
- [ ] I have verified the issue still exists in the `develop` branch of WPCS.

**Comment by jrfnl** [comment]
@peterwilsoncc I agree this can be annoying, though unless we add a never-ending list of exceptions, the sniff can not easily determine whether something is an intentional misspelling or an accidental one.

What do you suggest as a solution direction ?

**Comment by peterwilsoncc** [comment]
I've been trying to think of something that doesn't involve a bunch of exceptions an been unable too. 

The exceptions I was thinking of were:
* A string containing only a lowercase WordPress, ie `'wordpress'`. `'Wordpress'` continues to be replaced
* `define()` statements
* domains, with or without the scheme set

Each of these seem more likely to be settings rather than strings.

**Comment by jrfnl** [comment]
@peterwilsoncc Thanks for your input.

My thoughts:
1. I think blanket excepting lowercase `wordpress` goes a bit far, especially as this sniff has been available for quite a while now (good two years) and this is the first such reported issue.
2. I agree that it would probably be a good idea to exempt constant declarations from this sniff, both if declared using `define()` as well as using the `const` keyword.
3. The sniff already contains a regex which prevents any matches on `wordpress.org/com/net/tv`.

So for now, I think point 2 is the only one which needs action.

**Comment by jrfnl** [comment]
@peterwilsoncc Would you be able to test PR #1699 ?
